daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/transaction/TransactionSpec.scala

```diff
@@ -374,6 +374,97 @@ class TransactionSpec extends AnyFreeSpec with Matchers with ScalaCheckDrivenPro
       builder.build().contractKeys shouldBe expectedResults
     }
   }
+
+  "consumedContract" - {
+    "return the IDs of consumed contracts" in {
+      val builder = TransactionBuilder()
+      val parties = Seq("Alice")
+      def create() = {
+        val cid = builder.newCid
+        val node = builder.create(
+          cid,
+          template = "-pkg-:Mod:T",
+          argument = V.ValueUnit,
+          signatories = parties,
+          observers = Seq(),
+          key = None,
+        )
+        (cid, node)
+      }
+      def exercise(create: Node.NodeCreate[ContractId], consuming: Boolean) =
+        builder.exercise(
+          contract = create,
+          choice = "C",
+          actingParties = parties.toSet,
+          consuming = consuming,
+          argument = V.ValueUnit,
+          byKey = false,
+        )
```

Comment:
Perhaps it's possible to share these utilities between test cases? That would also make them shorter and so more readable.

Reply:
Factored out in 60a00a51a2b598954a75da07efc7de1f3e8c9fe3. Probably could do some more aggressive factoring but I’ll leave that for a future PR.